stripe/src/main/java/com/stripe/android/CustomerSession.java

```diff
@@ -489,20 +669,23 @@ public void handleMessage(Message msg) {
                         }
                         break;
                     case SOURCE_RETRIEVED:
-                        if (messageObject instanceof Source && mSourceRetrievalListener != null) {
-                            mSourceRetrievalListener.onSourceRetrieved((Source) messageObject);
+                        if (messageObject instanceof Source) {
+                            Source retrievedSource = (Source) messageObject;
+                            broadcastCustomerSource(retrievedSource);
+                            if (mSourceRetrievalListener != null) {
+                                mSourceRetrievalListener.onSourceRetrieved((Source) messageObject);
+                                mSourceRetrievalListener = null;
+                            }
                         }
-
-                        // A source listener only listens once.
```

Comment:
nit: should we put this comment back? (or maybe put it higher up in the javadoc as part of how this works)?